src/patch-builtins.js

```diff
@@ -246,9 +246,34 @@ let htmlElementMixin = utils.extendAll({
       nativeBlur.call(this);
     }
   }
-});
+};
+
+for (const property of Object.getOwnPropertyNames(Document.prototype)) {
+  const split = property.split('on');
+
+  if (split[0] === '') {
+    Object.defineProperty(htmlElementMixin, property, {
+      /** @this {HTMLElement} */
+      set: function(fn) {
+        const shadyData = ensureShadyDataForNode(this);
+        shadyData.__onCallbackListeners[property] && this.removeEventListener(split[1], shadyData.__onCallbackListeners[property]);
```

Comment:
Using the `split[1]` as the event name is problematic since 'onselectionchange'.split('on') == ["", "selecti", "change"]

Reply:
Ah good call. Instead, I think I should array spread in the definition and then join the remainder on `on` again.